### PR TITLE
Added docs for TableDefinition #coloumns & #remove_column [ci skip]

### PR DESCRIPTION
**Pull request by sakshi-keepworks**

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb [hunk 1]
```diff
@@ -214,6 +214,7 @@ def initialize(types, name, temporary, options, as = nil)
         @name = name
       end
 
+      # Returns an array of ColumnDefinitions for the columns of the table
```
Comment (rafaelfranca) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Missing `.` at the end

Comment (sjain1107) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Done.

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb [hunk 2]
```diff
@@ -369,6 +370,8 @@ def column(name, type, options = {})
         self
       end
 
+      # Removes given column from the table.
```
Comment (senny) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Can we reword this to:

> removed the column +name+ from the table.

File: activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb [hunk 3]
```diff
@@ -214,6 +214,7 @@ def initialize(types, name, temporary, options, as = nil)
         @name = name
       end
 
+      # Returns an array of ColumnDefinitions for the columns of the table.
```
Comment (senny) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: This is a tricky one. If we document that this method returns `ColumnDefinition` objects a user expects to find documentation on `ColumnDefinition` to explore possible interactions. Currently that class and it's subclasses are `:nodoc:`. /cc @rafaelfranca @carlosantoniodasilva

Comment (carlosantoniodasilva) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: Yeah, I agree it's weird because `ColumnDefinition` is not documented, however many other places in this file comment about `ColumnDefinition` (including the next method, and `TableDefinition` class), and they are available in our docs. So I'm thinking it might be ok? Maybe we should doc `ColumnDefinition` instead - or review or docs to remove their mentions?

Nitpicking, but it's `ColumnDefinition`, without the `s`.

Comment (senny) [inline activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb]: @carlosantoniodasilva fair enough. I'm currently reworking how we reference classes and methods in our documentation. I'll apply this patch for now and see how to improve the situation in my branch.
